### PR TITLE
Add webpack and load pickr with webpack

### PR DESCRIPTION
**Pull request by pabuisson**
## tl;dr

**Add webpack & load pickr with webpack**

Kept the webpack config as minimal as possible, most files being only copied from `src/` to `dist/` with the exception of `popup.js` (which now imports pickr) and the pickr CSS files that is copied not from `src/`, but from `node_modules`.

## More details

- Added webpack & its config file
- Added package.json to handle webpack, its plugins and the pickr dependency
- Moved all source files to `src/` directory. Webpack will build to `dist/` directory
- Webpack build logic: all existing files are simply copied from `src/` to `/dist`. The only exception is `popup.js`. In this file, we need to import pickr and therefore it must go through webpack pipeline to do so. Therefore:
    - `popup.js` is the only entry point,
    - `popup.js` is excluded from the files copied by the `CopyPlugin`
    - to make sure the loaded CSS for pickr is always in the same version as the JS loaded, I decided not to rely on the CDN to load the CSS. Therefore, it's copied from the `node_modules`, and has its own rule in the config of `CopyPlugin`

## Development workflow with this setup

- Run `npm install` to get all packages
- Run `npm run watch` to have webpack start watching your files. All output files will be built in `./dist`
- As an alternative, you can use `npm run build` for a one-shot build
- That's pretty much it :) 

## Screenshot

With addon built with webpack and loaded through `about:debugging` in Firefox 96:

Capture d’écran 2021-11-17 à 21 31 09

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "prettierlichess",
+  "version": "1.0.0",
+  "description": "Here is the source code for both the chrome and firefox extension! It's simple but it's mine. Feel free to download, fork, use, and mess with it all you want. No attribution or anything required.",
+  "scripts": {
+    "clean-dist": "rm -rf dist/*",
```
Comment (Durush) [inline package.json:6]: Hey can we replace this with something a bit more os flexible? or we are both windows so literally just "rmdir /s /q dist" would be fine with us

**Review by Durush — CHANGES_REQUESTED** [review]
Hey first of all, an absolutely huge thanks from both Noah and myself. Simply cannot believe you took the time to implement this for us. 

Just a small note for you here in the review but otherwise we tested on our end and everything looks great!

File: package.json (hunk 1 above)
Comment (pabuisson) [reply package.json]: I've added a dedicated webpack plugin for this (commit e705edc ),  so that it should be flexible and allow developers to use it no matter if they use windows, or linux, or mac os. Works for me on Mac OS and should work on windows too. Feel free to give it a try and tell me how it goes. 

If it does not work, I'll simply use the `rmdir` command you just suggested.

Comment (Durush) [reply package.json]: hey looks good with that update just validated it locally. Going to merge this in and we'll try to get this rolled out to firefox again. thanks so much again!

**Review by Durush — APPROVED** [review]
